### PR TITLE
Preserve data 🎉 🚀 ✔️ 

### PR DESCRIPTION
**Pull request by Nizamuddin4493**
# Preserve data 🎉 🚀 ✔️ 

- Store data in files.
- Run a program using the command line.
- On exit app saves all the data (books, people, and rentals) into JSON files.
- On startup, the app loads all data from files.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -63,11 +83,19 @@ def create_a_person
 
     case input
     when '1'
+      # rubocop:disable Layout/LineLength
       student = @create_student.new_student
-      @persons << student
+      s_struct = Struct.new(name: student.name, id: student.id, className: student.class, parent_permission: student.parent_permission, age: student.age)
```
Comment (GzimAsani) [inline app.rb]: - [ ] It is throwing an error here when I am trying to create a new person kindly update it
image

File: persons.json [hunk 2]
```diff
@@ -0,0 +1 @@
+["{\"json_class\":\"OpenStruct\",\"t\":{\"name\":\"Pierre Jardim\",\"id\":901,\"className\":\"Student\",\"parent_permission\":{\"parent_permission\":false},\"age\":19}}", "{\"json_class\":\"OpenStruct\",\"t\":{\"name\":\"Teacher X\",\"id\":556,\"className\":\"Teacher\",\"specialization\":\"Astrophysics\",\"age\":0}}", "{\"json_class\":\"OpenStruct\",\"t\":{\"name\":\"Khan\",\"id\":463,\"className\":\"Teacher\",\"specialization\":\"Doctor\",\"age\":34}}", "{\"json_class\":\"OpenStruct\",\"t\":{\"name\":\"Naveed\",\"id\":93,\"className\":\"Student\",\"parent_permission\":{\"parent_permission\":true},\"age\":45}}", "{\"json_class\":\"OpenStruct\",\"t\":{\"name\":\"Hameed\",\"id\":46,\"className\":\"Teacher\",\"specialization\":\"coder\",\"age\":29}}"]
```
Comment (GzimAsani) [inline persons.json]: - [ ] If the persons.json file is empty the app crashes, kindly update it with something similar to 
```ruby
    return [] unless File.exists?("people.json")
```

image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello    @Nizamuddin4493  and @Aquinasmoriasi       :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your setting up jsons file, they work well  :+1:
 
 ## Required Changes 

- [ ] Check the comment below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @Nizamuddin4493, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Aquinasmoriasi** [comment]
Thank you.

**Pull request merged**